lib/puppet/provider/foreman_smartproxy/rest_v3.rb

```diff
@@ -24,7 +24,7 @@ def exists?
   end
 
   def create
-    post_data = {:smart_proxy => {:name => resource[:name], :url => resource[:url]}}.to_json
+    post_data = {:smart_proxy => {:name => resource[:name], :url => resource[:url], :instance_id => resource[:instance_id]}}.to_json
```

Comment:
I guess it will fail and break compatibility with Foreman which don't accept this parameter. We do keep compatibility with older versions so that users can first upgrade their Puppet modules and then the actual instance. I also use it to run git master on my Foreman 2.4 server to find problems.

How about we only pass the instance ID if it's not nil? That allows users to not pass an instance ID on older versions.